packages/rtmpdump.rb

```diff
@@ -20,20 +21,12 @@ class Rtmpdump < Package
         i686: '109fb6fa70409f8ba274fa452c2e04dc1ff3ba740f618525c188139c1ddce363',
       x86_64: '0212408c6faad92b5909d77188d8c27e68ce955f61dcb5597603303e31e601ce',
   })
-  
-  def self.patch
-    system "sed -i 's,prefix=/usr/local,prefix=#{CREW_PREFIX},' Makefile"
-    system "sed -i 's,prefix=/usr/local,prefix=#{CREW_PREFIX},' librtmp/Makefile"
-    system "sed -i 's,libdir=\$(prefix)/lib,libdir=#{CREW_LIB_PREFIX},' librtmp/Makefile"
-    # OpenSSL builds are broken.
-    system "sed -e 's/^CRYPTO=OPENSSL/#CRYPTO=OPENSSL/' -e 's/#CRYPTO=GNUTLS/CRYPTO=GNUTLS/' -i Makefile -i librtmp/Makefile"
-  end
- 
+
   def self.build
-    system 'make'
+    system 'make', *@make_common_opts
   end
 
   def self.install
-    system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
+    system 'make', *@make_common_opts, "DESTDIR=#{CREW_DEST_DIR}", 'install'
```

Comment:
Shouldn't need these extra options.

Reply:
The full install paths are formed by concatenating  `DESTDIR` with variables that depend on `prefix`, so we do need to specify `prefix` and `libdir` here. In included `CRYPTO` in `@make_common_opts` because I didn't want to bother tracking down the details of how it's used. It may well suffice to put it in `self.build` only.

Reply:
@uberhacker Let me know if you have additional questions :smiley:

Reply:
I already made my peace.  We shouldn't need this.  Humor me and try it without and see what happens.

Reply:
**Attempt 1: Using `#{CREW_OPTIONS}` Only in `self.build`**
```ruby
# self.build
system "make #{CREW_OPTIONS}"
```
**Result 1** 
The build fails. The options in `CREW_OPTIONS` are valid only for configuration scripts generated by tools like `autoconf`.

**Attempt 2: Using `PREFIX=... LIBDIR=...` Only in `self.build`**
```ruby
# self.build
system "make PREFIX=/usr/local LIBDIR=/usr/local/lib64 CRYPTO=GNUTLS"
```
```ruby
# self.install
system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
```
**Result 2**
The build copies libraries to `/usr/local/lib` instead of `/usr/local/lib64`. Note that using `PREFIX` and `LIBDIR` in `self.install` would result in the output files being written directly to `/usr/local`, which is very much not what we want.

**Explanation**
If you look at the contents of `librtmp/Makefile` in the source, you can see that `libdir` and `prefix` are the variables used by the Makefile. `prefix` is not 100% necessary, because it is defaulted to `/usr/local` but I'm of the opinion that it's better to be explicit for something like that. Going with that thinking, I've added `SYS=posix` to the arguments as well. That variable is also used by the `Makefile`.
```make
# librtmp/Makefile
# line 3
prefix=/usr/local

# line 7
libdir=$(prefix)/lib

# line 11
LIBDIR=$(DESTDIR)$(libdir)

# line 51
SODIR_posix=$(LIBDIR)
SODIR=$(SODIR_$(SYS))

# line 117
install_so: librtmp$(SO_EXT)
	cp librtmp$(SO_EXT) $(SODIR)
```

Reply:
OK, so it looks like the `def self.patch` section is needed.  What about this is different from the previous build?  If it is, we would need to bump the version.

Reply:
There shouldn't be anything different (in terms of the result) from the previous build. This is just eliminating `self.patch` and instead passing the appropriate variables to `make` when we run it. I just have a purist mindset that was rubbed the wrong way by the idea of patching a file and calling out to external programs when all we really need to do is call `make` with the right arguments. 😅

Reply:
Fair enough.  With that said, it appears the project maintainer strayed from the beaten path with their Makefile. :)